test/test.js

```diff
@@ -470,6 +470,34 @@ exports['async next from consumer'] = function (test) {
     });
 };
 
+exports['generator throws error if next called after nil'] = function(test) {
+    var s = _(function(push, next) {
```

Comment:
As a general stylistic thing, can you put a `test.expect(1)` here? Same with the other test.